src/console/dlt-receive.c

```diff
@@ -614,15 +626,26 @@ int main(int argc, char *argv[])
     else
         dlt_set_id(dltdata.ecuid, DLT_RECEIVE_ECU_ID);
 
-    /* Connect to TCP socket or open serial device */
-    if (dlt_client_connect(&dltclient, dltdata.vflag) != DLT_RETURN_ERROR) {
+    do {
+        /* Connect to TCP socket or open serial device */
+        if (dlt_client_connect(&dltclient, dltdata.vflag) != DLT_RETURN_ERROR) {
 
-        /* Dlt Client Main Loop */
-        dlt_client_main_loop(&dltclient, &dltdata, dltdata.vflag);
+            /* Dlt Client Main Loop */
+            dlt_client_main_loop(&dltclient, &dltdata, dltdata.vflag);
 
-        /* Dlt Client Cleanup */
-        dlt_client_cleanup(&dltclient, dltdata.vflag);
-    }
+            if (dltdata.rflag == 1 && sig_close_recv == false) {
+                // if reconnect flag is enabled, wait for mentioned milli
+                // seconds
+                poll(NULL, 0, dltdata.rvalue);
+                continue;
+            }
+            else {
+                /* Dlt Client Cleanup */
+                dlt_client_cleanup(&dltclient, dltdata.vflag);
+                break;
+            }
+        }
+    } while (dltdata.rflag == 1 && poll(NULL, 0, dltdata.rvalue) == 0);
```

Comment:
this is somehow duplicated to the poll instruction before. You could use a while(1) instead.

I would prefer sleep() or nanosleep() instead of the poll to implement a sleep function.